lua/luasnip/session/snippet_collection.lua

```diff
@@ -250,12 +256,24 @@ function M.add_snippets(snippets, opts)
 	end
 end
 
+-- TODO benchmark without doing a shallow copy (is it worth to be
+-- unconventional and return tables with weird metatables to the outside?)
+-- TODO shallow copy functionality in nvim (I found nothing)?
+-- if not this should at least be moved to some utils file
```

Comment:
I agree, `luasnip.util.util`'s the place :D
Otherwise there's `vim.list_slice`, but that name isn't very nice.

Maybe specify this function is for lists, that way `ipairs` can be used instead of `pairs`.

Reply:
`ipairs` is only possible in the former case (`by_ft[type][ft]`), in the latter one (`by_ft[type]`) the returned table isn't a list but a dictionary.
As we already differ the two cases, we can of course use different copy functions.

Reply:
True!
Ah, also: if the whole table should be returned, a second copy of the snippets for each filetype becomes necessary, if we really don't want to leak the metatables.

Reply:
Oh that's right as well :see_no_evil:
In that case we need different copy functions either way, I think.